src/worker.cpp

```diff
@@ -1185,4 +1196,274 @@ void Worker::collectFrusFromJson()
+            i_source2VpdMap.find(l_recordName) == i_source2VpdMap.end())
+        {
+            logging::logMessage(
+                "Record: " + l_recordName +
+                ", is not found in the source2 path: " + i_source2Path);
+            continue;
+        }
+
+        for (const auto& l_keywordInfo : l_recordKwInfo.second)
+        {
+            const std::string& l_keyword = get<0>(l_keywordInfo);
+            const auto& l_defaultBinaryValue = get<1>(l_keywordInfo);
+            bool l_isPelRequired = get<2>(l_keywordInfo);
+
+            types::BinaryVector l_source1BinaryValue;
+            std::string l_source1StrValue;
+            if (!io_source1VpdMap.empty())
+            {
+                utils::getKwVal(io_source1VpdMap.at(l_recordName), l_keyword,
+                                l_source1StrValue);
+                l_source1BinaryValue = types::BinaryVector(
+                    l_source1StrValue.begin(), l_source1StrValue.end());
+            }
+            else
+            {
+                // Read keyword value from DBus
+                // Uncomment when readKeyword API implemenation is done
+                /* l_source1BinaryValue = utils::readKeyword(i_source1Path,
+                * std::tuple<types::Record, types::Keyword>(l_recordName,
+                l_keyword)); l_source1StrValue =
+                std::string(l_source1BinaryValue.begin(),
+                l_source1BinaryValue.end());
+                */
+            }
+
+            types::BinaryVector l_source2BinaryValue;
+            std::string l_source2StrValue;
+            if (!i_source2VpdMap.empty())
+            {
+                utils::getKwVal(i_source2VpdMap.at(l_recordName), l_keyword,
+                                l_source2StrValue);
+                l_source2BinaryValue = types::BinaryVector(
+                    l_source2StrValue.begin(), l_source2StrValue.end());
+            }
+            else
+            {
+                // Read keyword value from DBus
+                // Uncomment when readKeyword API implemenation is done
+                /* l_source2BinaryValue = utils::readKeyword(i_source2Path,
+                * std::tuple<types::Record, types::Keyword>(l_recordName,
+                l_keyword)); l_source2StrValue =
+                std::string(l_source2BinaryValue.begin(),
+                l_source2BinaryValue.end());
+                */
+            }
+
+            if (l_source1BinaryValue == l_defaultBinaryValue &&
+                l_source2BinaryValue == l_defaultBinaryValue)
+            {
+                if (l_isPelRequired)
+                {
+                    logging::logMessage(
+                        "Default value found in the source1 and source2 VPD, for record: " +
+                        l_recordName + " and keyword: " + l_keyword);
+                }
+                continue;
+            }
+            else if (l_source1BinaryValue == l_defaultBinaryValue &&
```

Comment:
Conditions are repeated so can be clubbed? Also, assuming write does update on DBus as well.

```
if(l_source1BinaryValue != l_source2BinaryValue)

{
    if(l_source2BinaryValue == l_defaultBinaryValue)
    {
        // TODO: Uncomment when writeKeyword API implemenation goes in,
        // handle if separate update to hardware and cache is required.
        // utils::writeKeyword(i_source2Path,
        // types::IpzData(l_recordName, l_keyword,
        // l_source1BinaryValue));
        continue;
    }
    
    io_source1VpdMap[l_recordName][l_keyword] = l_source2StrValue;
    
    if(l_source1BinaryValue == l_defaultBinaryValue)
    {
        // TODO: Uncomment when writeKeyword API implemenation goes in,
        // handle if separate update to hardware and cache is required.
        // utils::writeKeyword(i_source1Path,
        // types::IpzData(l_recordName, l_keyword,
        // l_source2BinaryValue));
    
        // Update map to publish the same data on DBus, which is already
        // present on the DBus.
        continue;
    }
    
    //Log PEL as both are not equal and not default.
}
else if(l_source1BinaryValue == l_defaultBinaryValue)
{
    if (l_isPelRequired)
    {
        logging::logMessage(
            "Default value found in the source1 and source2 VPD, for record: " +
            l_recordName + " and keyword: " + l_keyword);
    }
    continue;
}
```

Reply:
Done